src/shared/components/structureViewer/StructureVisualizer3D.ts

```diff
@@ -0,0 +1,325 @@
+            // frames: true
+        };
+
+        if (this._3dMolViewer) {
+            // clear previous content
+            this._3dMolViewer.clear();
+
+            $3Dmol.download(`pdb:${pdbId.toUpperCase()}`, this._3dMolViewer, options, () => {
+                this.updateViewer(chainId, residues, props);
+            });
+        }
+    }
+
+    public updateViewer(chainId:string,
+                        residues: IResidueSpec[] = this._residues,
+                        props:IStructureVisualizerProps = this._props)
+    {
+        this._props = props;
+        this._chainId = chainId;
+        this._residues = residues;
+
+        this.updateVisualStyle(residues, chainId, props);
+
+        this._3dMolViewer.render();
+    }
+
+    public updateResidues(residues: IResidueSpec[])
+    {
+        this._residues = residues;
+        this.updateViewer(this._chainId, residues);
+    }
+
+    protected selectAll()
+    {
+        this._selector = {};
+    }
+
+    protected selectChain(chainId: string)
+    {
+        this._selector = {chain: chainId};
+    }
+
+    protected setScheme(scheme: ProteinScheme)
+    {
+        this._style = StructureVisualizer3D.PROTEIN_SCHEME_PRESETS[scheme];
+        this.applyStyleForSelector();
+    }
+
+    protected setColor(color: string)
+    {
+        // save the color selection
+        this._color = this.formatColor(color);
+
+        if (this._style) {
+            // update current style with color information
+            _.each(this._style, (ele: AtomStyleSpec) => {
+                ele.color = this._color;
+            });
+        }
+
+        this.applyStyleForSelector();
+    }
+
+    protected formatColor(color: string)
+    {
+        // this is for 3Dmol.js compatibility
+        // (colors should start with an "0x" instead of "#")
+        return color.replace("#", "0x");
+    }
+
+
+    protected setTransparency(transparency:number)
+    {
+        _.each(this._style, (ele: AtomStyleSpec) => {
+            ele.opacity = (10 - transparency) / 10;
+        });
+
+        this.applyStyleForSelector();
+    }
+
+    protected rainbowColor()
+    {
+        this.setColor("spectrum");
+    }
+
+    protected cpkColor()
+    {
+        _.each(this._style, (ele: AtomStyleSpec) => {
+            // remove previous single color
+            delete ele.color;
+
+            // add default color scheme
+            ele.colors = $3Dmol.elementColors.defaultColors;
+        });
+
+        this.applyStyleForSelector();
+    }
+
+    protected selectAlphaHelix(chainId: string)
+    {
+        this._selector = {chain: chainId, ss: "h"};
+    }
+
+    protected selectBetaSheet(chainId: string)
+    {
+        this._selector = {chain: chainId, ss: "s"};
+    }
+
+    protected hideBoundMolecules()
+    {
+        // since there is no built-in "restrict protein" command,
+        // we need to select all non-protein structure...
+        const selector = {
+            resn: [
```

Comment:
According to documentation (http://3dmol.csb.pitt.edu/doc/types.html#AtomSpec), it is `Parent residue name`.

Actually, this is copied from the legacy code base. This selector is supposed to select all non-protein structure, that's all I remember...